crates/context_attribute/src/lib.rs

```diff
@@ -56,6 +56,11 @@ pub fn context(_attributes: TokenStream, input: TokenStream) -> TokenStream {
                     );
                 };
 
+                if field_type.as_str() != "MainOutput" {
+                    if let Visibility::Public(vis) = &field.vis {
+                        abort!(vis.pub_token, "context fields should not be public");
+                    }
+                }
```

Comment:
I am not sure about this being an error...

Reply:
After discussion, I removed the check. This branch now only contains the removal of `pub` and unused context fields.